chain/epoch-manager/src/lib.rs

```diff
@@ -430,12 +430,13 @@ impl EpochManager {
                                 stats.block_stats.expected as i64,
                             )) / 2
                         };
-                    (account, production_ratio)
+                    (production_ratio, account)
                 })
                 .collect::<Vec<_>>();
-            sorted_validators.sort_by_key(|a| a.1);
+            sorted_validators.sort();
```

Comment:
This is very neat fix!  

I think technically it is a bit unfair - in case of a draw the account id shouldn't be deciding who gets kicked out or not. To make it fair we could deterministically shuffle the validators using the chain randomness e.g. sorting by hash(account_id || vrf) instead of just by account id. In practice this doesn't matter as it should be a very rare occurrence.

Reply:
Yea I agree it's a little sad... but actually it will not be the only place where a validator's account ID breaks ties. It looks like we already do this when deciding who gets to be a block producer if there are ties in stakes. https://github.com/near/nearcore/blob/7cb32f1b7c28f88f5b4c18b645915059584c31af/chain/epoch-manager/src/validator_selection.rs#L66

there `order_proposals()` orders them by stake and breaks ties with account IDs. It's probably super unlikely to get ties in stake though

Reply:
Yeah, it's not a biggie, this is a very rare occurrence anyway.